dotnet/src/Connectors/VectorData.Abstractions/ConnectorSupport/VectorStoreRecordPropertyModel.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+
+namespace Microsoft.Extensions.VectorData.ConnectorSupport;
+
+/// <summary>
+/// Represents a property on a vector store record.
+/// </summary>
+[Experimental("MEVD9001")]
+public abstract class VectorStoreRecordPropertyModel(string modelName, Type clrType)
+{
+    private string? _storageName;
+
+    /// <summary>
+    /// The model name of the property. If the property corresponds to a .NET property, this name is the name of that property.
+    /// </summary>
+    public string ModelName { get; set; } = modelName;
+
+    /// <summary>
+    /// The storage name of the property. This is the name to which the property is mapped in the vector store.
+    /// </summary>
+    public string StorageName
+    {
+        get => this._storageName ?? this.ModelName;
+        set => this._storageName = value;
+    }
+
+    // See comment in VectorStoreJsonModelBuilder
+    // TODO: Spend more time thinking about this, there may be a less hacky way to handle it.
+
+    /// <summary>
+    /// A temporary storage name for the property, for use during the serialization process by certain connectors.
```

Comment:
Do we really need a temporary storage name? I just wonder if we are not exposing  something that should not exist in the first place.

Reply:
This is definitely a bit of hack, unfortunately. I hope we can remove it, but for now it's necessary for connectors which use an external serializer (JsonSerializer), and need to then manipulate the output ot the serializer before sending the results to the database.

In effect, a property has three possible names:
* The model name - CLR property name, e.g. HotelId
* The storage name - name appearing in the database, and used in filters. In some databases, the key **must** be called e.g. `id`
* The temporary storage name, which would be the name after JsonSerializer serialized the POCO; the crucially includes naming policy stuff, so HoteId in CamelCase is hotelId. This is what gets replaced by the connector with `id`.

Reply:
Thanks for the explanation. Lets merge it as is, but keep it in the back of our heads so one day we can somehow remove it.